Main/ClientApp/src/Components/Drawer/DrawerItemNested.tsx

```diff
@@ -25,35 +26,46 @@ const useStyles = makeStyles((theme: Theme) =>
 type DrawerItemNestedProps = {
   title: string;
   icon?: ReactElement;
-  menuItems: ReactElement;
+  menuItems?: ReactElement;
   link: string;
 };
 
 export default function DrawerItemNested(props: DrawerItemNestedProps) {
   const classes = useStyles();
   const [anchorEl, setAnchorEl] = useState<Element | null>(null);
+  const [redirectToCollection, setRedirectToCollection] = useState<boolean>(
+    false
+  );
   const onMenuOpen = (event: MouseEvent) => {
-    setAnchorEl(event.currentTarget as Element);
+    if (props.menuItems) setAnchorEl(event.currentTarget as Element);
   };
   const onMenuClose = () => setAnchorEl(null);
+
+  useEffect(() => {
+    setRedirectToCollection(false);
+  }, [redirectToCollection]);
+
```

Comment:
TBH, I have no idea what is going on here, but I'll give you the benefit of the doubt ;)

Reply:
you want to render Redirect component if user clicks ListItemButton and after rerender you want to reset bool value, so that button can be used again (without this reset button works only once)